app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodsscheduled/UploadMrnListController.scala

```diff
@@ -0,0 +1,161 @@
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.config.ViewConfig
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.connectors.UploadDocumentsConnector
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.controllers.JourneyControllerComponents
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.Nonce
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.UploadMrnListCallback
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.UploadDocumentsSessionConfig
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.views.html.rejectedgoods.upload_mrn_list_description
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class UploadMrnListController @Inject() (
+  val jcc: JourneyControllerComponents,
+  uploadDocumentsConnector: UploadDocumentsConnector,
+  val uploadDocumentsConfig: UploadDocumentsConfig,
+  val fileUploadConfig: FileUploadConfig,
+  val upload_mrn_list_description: upload_mrn_list_description
+)(implicit val ec: ExecutionContext, val appConfig: ViewConfig)
+    extends RejectedGoodsScheduledJourneyBaseController {
+
+  final val continueUrl: Call    = routes.CheckClaimantDetailsController.show()
+  final val backlinkUrl: Call    = routes.CheckDeclarationDetailsController.show()
+  final val callbackAction: Call = routes.UploadMrnListController.submit()
+  final val selfUrl: String      = jcc.servicesConfig.getString("self.url")
+
+  final val show: Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    uploadDocumentsConnector
+      .initialize(
+        UploadDocumentsConnector
+          .Request(
+            uploadDocumentsSessionConfig(journey.answers.nonce),
+            journey.answers.scheduledDocument.map(file => Seq(file)).getOrElse(Seq.empty)
+          )
+      )
+      .map {
+        case Some(url) =>
+          Redirect(s"${uploadDocumentsConfig.publicUrl}$url")
+        case None      =>
+          Redirect(
+            s"${uploadDocumentsConfig.publicUrl}${uploadDocumentsConfig.contextPath}"
+          )
+      }
+  }
+
+  @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
+  final val submit: Action[AnyContent] = simpleActionReadWriteJourney(
+    { implicit request => journey =>
+      request
+        .asInstanceOf[Request[AnyContent]]
+        .body
+        .asJson
+        .flatMap(_.asOpt[UploadMrnListCallback]) match {
+        case None =>
+          logger.warn("missing or invalid callback payload")
+          (journey, BadRequest("missing or invalid callback payload"))
+
+        case Some(callback) =>
+          callback.uploadedFiles.headOption match {
+            case Some(uploadedFile) =>
+              journey
+                .receiveScheduledDocument(
+                  callback.nonce,
+                  uploadedFile
+                )
+                .fold(
+                  error => (journey, BadRequest(error)),
+                  modifiedJourney => (modifiedJourney, NoContent)
```

Comment:
Shouldn't this redirect to `routes.CheckClaimantDetailsController.show()`

Reply:
@airudah2000 this is a callback endpoint, it is called by the external microservice once the file is uploaded and removed